### ISSUE TITLE
ruamel.yaml.clib and mac m1 (arm 64)

### ISSUE THREAD

**Issue by denadai2** [pre-commit/pre-commit-hooks#804]
I have an error that happens only when I use pre-commit hooks in my `.pre-commit-config.yaml`:

```
repos:
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.3.0
    hooks:
    -   id: check-merge-conflict
-   repo: https://github.com/psf/black
    rev: 22.3.0
    hooks:
    -   id: black
 ```

The **error** is:

```
pre-commit run --all-files
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/psf/black.
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/Users/mdenadai/.cache/pre-commit/repotnoz8qjc/py_env-python3.9/bin/python', '-mpip', 'install', '.')
return code: 1
expected return code: 0
stdout:
    Processing /Users/mdenadai/.cache/pre-commit/repotnoz8qjc
      Preparing metadata (setup.py): started
      Preparing metadata (setup.py): finished with status 'done'
    Collecting ruamel.yaml>=0.15
      Using cached ruamel.yaml-0.17.21-py3-none-any.whl (109 kB)
    Collecting tomli>=1.1.0
      Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
    Collecting ruamel.yaml.clib>=0.2.6
      Using cached ruamel.yaml.clib-0.2.6.tar.gz (180 kB)
      Preparing metadata (setup.py): started
      Preparing metadata (setup.py): finished with status 'error'

stderr:
      error: subprocess-exited-with-error

      × python setup.py egg_info did not run successfully.
      │ exit code: 1
      ╰─> [9 lines of output]
          ld: warning: ignoring file /var/folders/xf/smz7c4096mv2sqcvw83b989h0000gp/T/tmp_ruamel_ympprlef/var/folders/xf/smz7c4096mv2sqcvw83b989h0000gp/T/tmp_ruamel_ympprlef/test_ruamel_yaml.o, building for macOS-x86_64 but attempting to link with file built for unknown-arm64
          Undefined symbols for architecture x86_64:
            "_main", referenced from:
               implicit entry/start for main executable
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          sys.argv ['/private/var/folders/xf/smz7c4096mv2sqcvw83b989h0000gp/T/pip-install-4gks94rl/ruamel-yaml-clib_715544d7a0ac445babaadeebc663b25a/setup.py', 'egg_info', '--egg-base', '/private/var/folders/xf/smz7c4096mv2sqcvw83b989h0000gp/T/pip-pip-egg-info-tqe1raku']
          test compiling /var/folders/xf/smz7c4096mv2sqcvw83b989h0000gp/T/tmp_ruamel_ympprlef/test_ruamel_yaml.c -> test_ruamel_yaml link error /var/folders/xf/smz7c4096mv2sqcvw83b989h0000gp/T/tmp_ruamel_ympprlef/test_ruamel_yaml.c
          Exception: command '/usr/bin/clang' failed with exit code 1
          [end of output]

      note: This error originates from a subprocess, and is likely not a problem with pip.
    error: metadata-generation-failed

    × Encountered error while generating package metadata.
    ╰─> See above for output.

    note: This is an issue with the package mentioned above, not pip.
    hint: See above for details.
```

pre-commit 2.20.0. As the title says, I have an Apple M1

**Comment by asottile** [comment]
this has nothing to do with pre-commit-hooks -- please report your bug to the actually broken library

but it also looks like you're using an x86_64 python on m1 so that's probably your issue

**Comment by gmeans** [comment]
tl:dr; This error appears when using any of the `pre-commit-hooks`, not just `check-yaml`. I'm also not sure the issue is solely w/ `ruamel.yaml`.

The issue appears to be due to the install looking for the python dev headers, but is unable to find them. I think this is due to the header path location changing w/ the latest Xcode upgrade.

```
# from the error message below
/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Headers

# the path on my machine
/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/Headers
```

I'm not sure how Python determines the paths to use for this. FWIW I'm able to install both `ruamel.yaml` and `ruamel.yaml.clib`. Does pre-commit do anything to attempt to pass these values to the hooks as they install?

Thanks!

## Ran using this config (Sample Config minus the check-yaml)
```
# See https://pre-commit.com for more information
# See https://pre-commit.com/hooks.html for more hooks
repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.3.0
    hooks:
      -   id: trailing-whitespace
      -   id: end-of-file-fixer
      # -   id: check-yaml
      -   id: check-added-large-files
```

## Error received

```
❯ pre-commit                     
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/Users/meansg/.cache/pre-commit/repopc_p50b0/py_env-python3.9/bin/python', '-mpip', 'install', '.')
return code: 1
expected return code: 0
stdout:
    Processing /Users/meansg/.cache/pre-commit/repopc_p50b0
      Preparing metadata (setup.py): started
      Preparing metadata (setup.py): finished with status 'done'
    Collecting ruamel.yaml>=0.15
      Using cached ruamel.yaml-0.17.21-py3-none-any.whl (109 kB)
    Collecting tomli>=1.1.0
      Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
    Collecting ruamel.yaml.clib>=0.2.6
      Using cached ruamel.yaml.clib-0.2.6.tar.gz (180 kB)
      Preparing metadata (setup.py): started
      Preparing metadata (setup.py): finished with status 'done'
    Building wheels for collected packages: pre-commit-hooks, ruamel.yaml.clib
      Building wheel for pre-commit-hooks (setup.py): started
      Building wheel for pre-commit-hooks (setup.py): finished with status 'done'
      Created wheel for pre-commit-hooks: filename=pre_commit_hooks-4.3.0-py2.py3-none-any.whl size=40807 sha256=2f6fa9a364370deff5988ef2a5988fdabb94026f04a1e63211428a2e3835151d
      Stored in directory: /private/var/folders/vr/yj5q1psn0zq_54hhr9k4y7780000gp/T/pip-ephem-wheel-cache-itv3jbr4/wheels/e9/83/08/7127193b44ee73db5aa8ebfc42e145ac91de740522a0f55d1c
      Building wheel for ruamel.yaml.clib (setup.py): started
      Building wheel for ruamel.yaml.clib (setup.py): finished with status 'error'
      Running setup.py clean for ruamel.yaml.clib
    Successfully built pre-commit-hooks
    Failed to build ruamel.yaml.clib
    Installing collected packages: tomli, ruamel.yaml.clib, ruamel.yaml, pre-commit-hooks
      Running setup.py install for ruamel.yaml.clib: started
      Running setup.py install for ruamel.yaml.clib: finished with status 'error'
    
stderr:
      error: subprocess-exited-with-error
      
      × python setup.py bdist_wheel did not run successfully.
      │ exit code: 1
      ╰─> [25 lines of output]
          sys.argv ['/private/var/folders/vr/yj5q1psn0zq_54hhr9k4y7780000gp/T/pip-install-v96xpvqk/ruamel-yaml-clib_b596bf47b4914aaaa404eac657425266/setup.py', 'bdist_wheel', '-d', '/private/var/folders/vr/yj5q1psn0zq_54hhr9k4y7780000gp/T/pip-wheel-m3vde13u']
          test compiling /var/folders/vr/yj5q1psn0zq_54hhr9k4y7780000gp/T/tmp_ruamel_5q_yerp0/test_ruamel_yaml.c -> test_ruamel_yaml OK
          /Users/meansg/.cache/pre-commit/repopc_p50b0/py_env-python3.9/lib/python3.9/site-packages/setuptools/dist.py:286: SetuptoolsDeprecationWarning: The namespace_packages parameter is deprecated, consider using implicit namespaces instead (PEP 420).
            warnings.warn(msg, SetuptoolsDeprecationWarning)
          running bdist_wheel
          running build
          running build_py
          creating build
          creating build/lib.macosx-10.9-universal2-cpython-39
          creating build/lib.macosx-10.9-universal2-cpython-39/ruamel
          copying .ruamel/__init__.py -> build/lib.macosx-10.9-universal2-cpython-39/ruamel
          creating build/lib.macosx-10.9-universal2-cpython-39/ruamel/yaml
          copying .ruamel/yaml/__init__.py -> build/lib.macosx-10.9-universal2-cpython-39/ruamel/yaml
          creating build/lib.macosx-10.9-universal2-cpython-39/ruamel/yaml/clib
          copying ./__init__.py -> build/lib.macosx-10.9-universal2-cpython-39/ruamel/yaml/clib
          copying ./LICENSE -> build/lib.macosx-10.9-universal2-cpython-39/ruamel/yaml/clib
          running build_ext
          building '_ruamel_yaml' extension
          creating build/temp.macosx-10.9-universal2-cpython-39
          clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Headers -arch arm64 -arch x86_64 -Werror=implicit-function-declaration -I/Users/meansg/.cache/pre-commit/repopc_p50b0/py_env-python3.9/include -I/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Headers -c _ruamel_yaml.c -o build/temp.macosx-10.9-universal2-cpython-39/_ruamel_yaml.o
          _ruamel_yaml.c:6:10: fatal error: 'Python.h' file not found
          #include "Python.h"
                   ^~~~~~~~~~
          1 error generated.
          error: command '/usr/bin/clang' failed with exit code 1
          [end of output]
      
      note: This error originates from a subprocess, and is likely not a problem with pip.
      ERROR: Failed building wheel for ruamel.yaml.clib
      error: subprocess-exited-with-error
      
      × Running setup.py install for ruamel.yaml.clib did not run successfully.
      │ exit code: 1
      ╰─> [27 lines of output]
          sys.argv ['/private/var/folders/vr/yj5q1psn0zq_54hhr9k4y7780000gp/T/pip-install-v96xpvqk/ruamel-yaml-clib_b596bf47b4914aaaa404eac657425266/setup.py', 'install', '--record', '/private/var/folders/vr/yj5q1psn0zq_54hhr9k4y7780000gp/T/pip-record-4z5x1p1h/install-record.txt', '--single-version-externally-managed', '--compile', '--install-headers', '/Users/meansg/.cache/pre-commit/repopc_p50b0/py_env-python3.9/include/site/python3.9/ruamel.yaml.clib']
          test compiling /var/folders/vr/yj5q1psn0zq_54hhr9k4y7780000gp/T/tmp_ruamel_o97xq7s8/test_ruamel_yaml.c -> test_ruamel_yaml OK
          /Users/meansg/.cache/pre-commit/repopc_p50b0/py_env-python3.9/lib/python3.9/site-packages/setuptools/dist.py:286: SetuptoolsDeprecationWarning: The namespace_packages parameter is deprecated, consider using implicit namespaces instead (PEP 420).
            warnings.warn(msg, SetuptoolsDeprecationWarning)
          running install
          /Users/meansg/.cache/pre-commit/repopc_p50b0/py_env-python3.9/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
            warnings.warn(
          running build
          running build_py
          creating build
          creating build/lib.macosx-10.9-universal2-cpython-39
          creating build/lib.macosx-10.9-universal2-cpython-39/ruamel
          copying .ruamel/__init__.py -> build/lib.macosx-10.9-universal2-cpython-39/ruamel
          creating build/lib.macosx-10.9-universal2-cpython-39/ruamel/yaml
          copying .ruamel/yaml/__init__.py -> build/lib.macosx-10.9-universal2-cpython-39/ruamel/yaml
          creating build/lib.macosx-10.9-universal2-cpython-39/ruamel/yaml/clib
          copying ./__init__.py -> build/lib.macosx-10.9-universal2-cpython-39/ruamel/yaml/clib
          copying ./LICENSE -> build/lib.macosx-10.9-universal2-cpython-39/ruamel/yaml/clib
          running build_ext
          building '_ruamel_yaml' extension
          creating build/temp.macosx-10.9-universal2-cpython-39
          clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Headers -arch arm64 -arch x86_64 -Werror=implicit-function-declaration -I/Users/meansg/.cache/pre-commit/repopc_p50b0/py_env-python3.9/include -I/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/Headers -c _ruamel_yaml.c -o build/temp.macosx-10.9-universal2-cpython-39/_ruamel_yaml.o
          _ruamel_yaml.c:6:10: fatal error: 'Python.h' file not found
          #include "Python.h"
                   ^~~~~~~~~~
          1 error generated.
          error: command '/usr/bin/clang' failed with exit code 1
          [end of output]
      
      note: This error originates from a subprocess, and is likely not a problem with pip.
    error: legacy-install-failure
    
    × Encountered error while trying to install package.
    ╰─> ruamel.yaml.clib
    
    note: This is an issue with the package mentioned above, not pip.
    hint: See above for output from the failure.
    
Check the log at /Users/meansg/.cache/pre-commit/pre-commit.log
```

**Comment by asottile** [comment]
as above, I can assure you this has nothing to do with pre-commit
